bug(rulegen):`thiserror` is broken in rulegen generated boilerplate

shorthand for interpolating fields from the error provided by [`thiserror`](https://docs.rs/thiserror/latest/thiserror/#details) crate seems not working in the generated boilerplate
```rust
// minimal reproducible example (boilerplate not included)
#[derive(Debug, Error)]
#[error("{0}")]
pub struct Foo (&'static str);
```
expected to pass the compilation, but the compiler yell at me:
```
   |
19 | #[derive(Debug, Error)]
   |                 ^^^^^ use of undeclared crate or module `thiserror`
   |
   = note: this error originates in the derive macro `Error` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `as_display` found for reference `&&'static str` in the current scope
   |
20 | #[error("{0}")]
   |         ^^^^^ method not found in `&&str`
   |

   |
5  |     fn as_display(&self) -> Self;
   |        ---------- the method is available for `&&'static str` here
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
1  + use oxc_diagnostics::thiserror::__private::DisplayAsDisplay;
```

--- a/tasks/rulegen/template.txt
+++ b/tasks/rulegen/template.txt
@@ -1,6 +1,6 @@
 use oxc_diagnostics::{
     miette::{self, Diagnostic},
-    thiserror::Error,
+    thiserror::{self, Error},
 };
 use oxc_macros::declare_oxc_lint;
 use oxc_span::Span;

